modules/publicize/publicize.php

```diff
@@ -137,14 +137,18 @@ function get_profile_link( $service_name, $c ) {
 		$cmeta = $this->get_connection_meta( $c );
 
 		if ( isset( $cmeta['connection_data']['meta']['link'] ) ) {
-			if ( 'facebook' == $service_name && 0 === strpos( parse_url( $cmeta['connection_data']['meta']['link'], PHP_URL_PATH ), '/app_scoped_user_id/' ) ) {
-				// App-scoped Facebook user IDs are not usable profile links
-				return false;
+			if ( 'facebook' != $service_name ) {
+				return $cmeta['connection_data']['meta']['link'];
+			}
+			if( 0 !== strpos( parse_url( $cmeta['connection_data']['meta']['link'], PHP_URL_PATH ), '/app_scoped_user_id/' ) ) {
+				return $cmeta['connection_data']['meta']['link'];
 			}
+		}
 
-			return $cmeta['connection_data']['meta']['link'];
-		} elseif ( 'facebook' == $service_name && isset( $cmeta['connection_data']['meta']['facebook_page'] ) ) {
```

Comment:
This is because a connection could have the `$cmeta['connection_data']['meta']['link']` set, but not match either of the `if` statements within that `if`.  When that is the case, then the function should continue to try and find a valid profile link.